dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java

```diff
@@ -136,18 +135,6 @@ protected void initBinder( WebDataBinder binder )
         this.enumClasses.forEach( c -> binder.registerCustomEditor( c, new ConvertEnum( c ) ) );
     }
 
-    @ExceptionHandler( org.hisp.dhis.feedback.BadRequestException.class )
-    @ResponseBody
-    public WebMessage badRequestException( org.hisp.dhis.feedback.BadRequestException ex )
-    {
-        WebMessage message = badRequest( ex.getMessage(), ex.getCode() );
-        if ( !ex.getErrorReports().isEmpty() )
-        {
-            message.setResponse( new ErrorReportsWebMessageResponse( ex.getErrorReports() ) );
-        }
-        return message;
-    }
-
```

Comment:
This needs to stay as the other handler below will not add the error reports like this does.

Reply:
Then I guess we need it for a ForbiddenException too, right?

Reply:
The https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/ForbiddenException.java cannot (yet) have error reports attached to it so currently it does not have this.

Reply:
That's what I meant, is there a reason why ForbiddenException doesn't generate the report?

Reply:
Oh, I added just what was needed to that point. If you want a `ForbiddenException` with a report you can add the field and a similar mapping to the `WebMessage` like I did for `BadRequestException`. But usually a forbidden does not need much of a detail that would come with a report. I guess that is why it wasn't needed so far.